src/SFA.DAS.EmployerIncentives.Database/Scripts/Generate.EmploymentCheckFailureEmailsReport.sql

```diff
@@ -2,40 +2,65 @@ DECLARE @PeriodNumber TINYINT
 DECLARE @PaymentYear SMALLINT
 
 -- Update period & payment year here
-SET @PeriodNumber = 12
-SET @PaymentYear = 2122
+SET @PeriodNumber = 2
+SET @PaymentYear = 2223
+
 -- Update period & payment year here
 
 DECLARE @EmployedAtStartOfApprenticeshipFailures TABLE 
 (
-	PendingPaymentId UNIQUEIDENTIFIER
+	uln bigint
 )
 
 DECLARE @EmployedBeforeSchemeStartedFailures TABLE 
 (
-	PendingPaymentId UNIQUEIDENTIFIER
+	uln bigint
 )
 
 -- Pending payments that have failed EC validation in previous runs
 INSERT INTO @EmployedAtStartOfApprenticeshipFailures
-SELECT DISTINCT PendingPaymentId 
-FROM incentives.PendingPaymentValidationResult
+SELECT DISTINCT ULN
+FROM incentives.PendingPaymentValidationResult ppvr
+INNER JOIN [incentives].[PendingPayment] pp On ppvr.PendingPaymentId = pp.Id
+INNER JOIN [incentives].[Learner] as l on pp.ApprenticeshipIncentiveID = l.ApprenticeshipIncentiveID
+WHERE Step = 'EmployedAtStartOfApprenticeship' 
+AND Result = 0 
+AND ppvr.PeriodNumber < @PeriodNumber 
+AND ppvr.PaymentYear = @PaymentYear
+union
+SELECT DISTINCT ULN
+FROM incentives.PendingPaymentValidationResult ppvr
+INNER JOIN [incentives].[PendingPayment] pp On ppvr.PendingPaymentId = pp.Id
+INNER JOIN [incentives].[Learner] as l on pp.ApprenticeshipIncentiveID = l.ApprenticeshipIncentiveID
 WHERE Step = 'EmployedAtStartOfApprenticeship' 
 AND Result = 0 
-AND PeriodNumber <> @PeriodNumber 
-OR PaymentYear <> @PaymentYear
+AND ppvr.PaymentYear < @PaymentYear
```

Comment:
This won't include payment validation failures for any previous periods in the current academic year

Reply:
this is union by another query above this and that will exclude any payment failures from current academic year, same for second list , line 28 & 29